cmd/manager/main.go

```diff
@@ -155,6 +160,16 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Setup health checks
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		log.Error(err, "Unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
```

Comment:
I understand why Keycloak and Postgres, registries have readiness probe

> Readiness
> Readiness probes are designed to let Kubernetes know when your app is ready to serve traffic. Kubernetes makes sure the readiness probe passes before allowing a service to send traffic to the pod. If a readiness probe starts to fail, Kubernetes stops sending traffic to the pod until it passes.

But as far as I understand Che Operator does not provide any REST API.
So, I'm missing some point or the readiness probe is redundant.

The readiness probe won't affect metrics, right https://github.com/eclipse/che-operator/blob/probes/deploy/olm-catalog/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml#L372 ?

Reply:
> But as far as I understand Che Operator does not provide any REST API.

`mgr.AddReadyzCheck("readyz", healthz.Ping)` does the trick.

> The readiness probe won't affect metrics, right

It shouldn't. Do you have any concerns ?